director/jobs/models.py

```diff
@@ -0,0 +1,433 @@
+from enum import unique
+import json
+
+from django_celery_beat.models import PeriodicTask
+from django.core import validators
+from django.db import models
+from django.utils import timezone
+from jsonfallback.fields import FallbackJSONField
+
+from lib.enum_choice import EnumChoice
+from accounts.models import Account
+from projects.models import Project
+from users.models import User
+
+
+class Zone(models.Model):
+    """
+    A zone within which jobs are run.
+
+    Zones are similar to, and may correspond with, "availability zones" as
+    provided by public providers such as AWS and Google Cloud. Workers
+    and the data they operate on are usually colated within zones to
+    reduce latency.
+
+    Stencila provides several zones, and users can decide which zone their
+    project will run in. Accounts can define their own zones and select those
+    for specific projects.
+    """
+
+    account = models.ForeignKey(
+        Account,
+        on_delete=models.CASCADE,
+        related_name="zones",
+        help_text="The account that this zone is linked to.",
+    )
+
+    name = models.CharField(
+        max_length=256,
+        validators=[
+            validators.RegexValidator(
+                r"^[a-z][a-z0-9\-]*$",
+                "Name should start with a lowercase letter and only contain lowercase letters, digits and hyphens",
+            )
+        ],
+        help_text="The identifier of the queue the job was posted to.",
+    )
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=["account", "name"], name="unique_name")
+        ]
+
+
+class Worker(models.Model):
+    """
+    A worker that runs jobs within a zone.
+
+    This model stores information on a worker as captured by the `overseer` service
+    from Celery's worker monitoring events.
+    """
+
+    zone = models.ForeignKey(
+        Zone,
+        null=True,
+        blank=True,
+        on_delete=models.CASCADE,
+        related_name="workers",
+        help_text="The zone that this worker is in.",
+    )
+
+    created = models.DateTimeField(
```

Comment:
General note and I think we've agreed on this already, `created`, `starteds`, `updated` would be `DateTimeFields`. If we wanted to indicate that a task has been started, updated or finished, use `is_started`, `is_updated`, `is_finished` etc as booleans.